components/script/dom/cssstyledeclaration.rs

```diff
@@ -115,10 +115,16 @@ impl CSSStyleOwner {
                 if changed {
                     // If this is changed, see also
                     // CSSStyleRule::SetSelectorText, which does the same thing.
-                    rule.global()
-                        .as_window()
-                        .Document()
-                        .invalidate_stylesheets();
+                    if let Some(shadow_root) =
+                        shadow_root_from_node(rule.parent_stylesheet().owner().upcast::<Node>())
```

Comment:
It seems less footgunny to write a function that returns the owner document or shadow root, wdyt?

Reply:
I ended up adding `invalidate_stylesheets` to the `StyleSheetsListOwner` trait and using the `stylesheets_owner_from_node` method instead.